manifests/params.pp

```diff
@@ -4,28 +4,32 @@
 # The python Module default configuration settings.
 #
 class python::params {
-  $ensure                    = 'present'
-  $version                   = 'system'
-  $pip                       = 'present'
-  $dev                       = 'absent'
-  $virtualenv                = 'absent'
-  $gunicorn                  = 'absent'
-  $manage_gunicorn           = true
-  $manage_python_package     = true
-  $manage_virtualenv_package = true
-  $manage_pip_package        = true
-  $provider                  = undef
-  $valid_versions            = undef
-  $manage_scl                = true
+  ## Module compatibility check
+  unless $facts['os']['family'] in [ 'Debian', 'RedHat', 'Suse', 'Gentoo', 'AIX', ] {
+    fail("Module is not compatible with ${facts['os']['name']}")
```

Comment:
i wouldn't do this
we have enough hooks into the module to use it on unsupported operating systems, but this makes it impossible

Reply:
This just has been moved from init.pp to params.pp to have the init.pp code cleaned. I wanted to keep it since I don't wanted to change too much.